### ISSUE TITLE
s3fs statically sets host header to s3.amazonaws.com in some cases

### ISSUE THREAD

**Issue by rvdm** [s3fs-fuse/s3fs-fuse#153]
Hi,

When using a non-Amazon hosted S3 bucket, s3fs adds a static host: header to its requests. This results in HTTP 400 errors from non-AWS S3 buckets, breaking s3fs on top of alternative S3 implementations.

From src/curl.cpp:
``` c++
requestHeaders = curl_slist_sort_insert(requestHeaders, "host", string(bucket + ".s3.amazonaws.com").c_str());
```

**Comment by andrewgaul** [comment]
Note that this is only done for V4 authentication.

**Comment by rvdm** [comment]
Correct, but the client will try V4 first (unless it's disabled as a mount option). Anyway, hardcoding hostnames is never a good idea :) 

I would love to write a patch, but my cpp is too rusty for it. More clients have this issue, and for folks hosting an S3 endpoint it's not a very crazy idea to just add the Amazon hostname(s) to the virtualhost configuration - but it's a better idea to just clean it up :)

**Comment by andrewgaul** [comment]
Fixed by #164.

**Comment by ggtakec** [comment]
@andrewgaul @kahing 
thank you.
